### ISSUE TITLE
Add support for GET requests

### ISSUE THREAD

**Issue by mythmon** [nearform/graphql-hooks#211]
### Description

Please add support for making requests to the GraphQL server via GET requests as well as POST requests. This is [described in the GraphQL docs](https://graphql.org/learn/serving-over-http/#get-request). The query details are included in the query parameters of the request, instead of the body.

### Suggested implementation

When configuration the client it would be nice to be able to simply specify a method to use, like this:

```js
const client = new GraphQLClient({
  url: 'https://example.com/graphql',
  method: 'GET',
});
```

**Comment by Joezo** [comment]
Would there be a situation where you might want to send one request via GET but another via POST? That might be tricky for us to do.

**Comment by mythmon** [comment]
A feature I've seen in some other GraphQL libraries is to use GET for queries and POST for mutations. That aligns well with the semantics of HTTP, and makes a lot of other systems, like caching, work nicely. So, using GET for some requests and POST for others would be helpful.

That being said, I'm personally only using this for queries, not mutations. Because of that, I'd be fine with having everything on a single method.

**Comment by bmullan91** [comment]
@mythmon Yep GET for queries, POST for mutations would be the ideal implementation - this also matches Apollo.

`apollo-http-link` docs:

> useGETForQueries: set to true to use the HTTP GET method for queries (but not for mutations)

@mythmon Do you fancy making a PR for this? 😄

**Comment by mythmon** [comment]
I might be able to hack on this next weekend, but I've moved over to Apollo and [react-apollo-hooks](https://www.npmjs.com/package/react-apollo-hooks). It would still be nice to have more options and more independent implementations though. I'll update this issue with any progress I do or don't make.

**Comment by joeynimu** [comment]
@bmullan91 I could give this a shot with some guidance :-D

**Comment by cbalano** [comment]
Setting the HTTP verb to get actually works. Like this:
const client  = new GraphQLClient({url: 'https://something.server.com/gql', fetchOptions: {mode:'no-cors', method:'GET'}

The showstopper arises after : (

TypeError: Failed to execute 'fetch' on 'Window': Request with GET/HEAD method cannot have body.
    at GraphQLClient.request (graphql-hooks.js:245)
    at graphql-hooks.js:430
    at graphql-hooks.js:496
    at commitHookEffectList (react-dom.development.js:21073)
    at commitPassiveHookEffects (react-dom.development.js:21106)
    at HTMLUnknownElement.callCallback (react-dom.development.js:363)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:412)
    at invokeGuardedCallback (react-dom.development.js:466)
    at flushPassiveEffectsImpl (react-dom.development.js:24196)
    at unstable_runWithPriority (scheduler.development.js:674)
    at runWithPriority$2 (react-dom.development.js:11834)
    at flushPassiveEffects (react-dom.development.js:24167)
    at react-dom.development.js:23728
    at scheduler_flushTaskAtPriority_Normal (scheduler.development.js:450)
    at flushTask (scheduler.development.js:503)
    at flushWork (scheduler.development.js:635)
    at MessagePort.performWorkUntilDeadline (scheduler.development.js:237)

**Comment by helloguille** [comment]
I have the same need:
Sometimes you need to cache the API response from a CDN, that's why you might want to use GET for non-mutation queries (CDNs like Google Cloud CDN will only cache GET requests)

**Comment by galmail** [comment]
yes makes sense.. I will try to raise a PR for that

**Comment by helloguille** [comment]
Hi, pull request author here, I changed a few lines to add support for GET method, via fetchOptionsOverrides: {method: "GET"}

**Comment by bmullan91** [comment]
Fixed in #410

**Comment by bmullan91** [comment]
Available in `graphql-hooks@4.1.0` 🚀
